puppetry/__init__.py

```diff
@@ -180,22 +181,34 @@ def sendSet(data):
     ''' Send a set request to the viewer
             data must be a dict
     '''
+
+    req_id = -1
     if _running:
         if isinstance(data, dict):
+            req_id = get_next_request_id()
             msg = { 'command':'set', 'data':data }
-            msg.setdefault('reqid', get_next_request_id())
+            msg.setdefault('reqid', req_id)
```

Comment:
Since `msg` is newly constructed on the previous line, we know it doesn't already have a `'reqid'` key, so this `setdefault()` will unconditionally assign. Put differently, looks to me as though line 189 could simply read:

    msg = { 'command':'set', 'data':data, 'reqid':req_id }